packages/core/src/interactions/packet/forward.ts

```diff
@@ -1,80 +1,55 @@
 import { PROTOCOL_STRING } from '../../constants'
 import { Packet } from '../../messages/packet'
-import { AcknowledgementMessage } from '../../messages/acknowledgement'
-import Debug from 'debug'
+import type HoprCoreEthereum from '@hoprnet/hopr-core-ethereum'
+// import Debug from 'debug'
 import type PeerId from 'peer-id'
-import { durations, oneAtATime, HoprDB } from '@hoprnet/hopr-utils'
+import { durations, pubKeyToPeerId, HoprDB } from '@hoprnet/hopr-utils'
 import { Mixer } from '../../mixer'
-import { Challenge } from '../../messages/packet/challenge'
-import { PROTOCOL_ACKNOWLEDGEMENT } from '../../constants'
-import LibP2P from 'libp2p'
-import HoprCoreEthereum from '@hoprnet/hopr-core-ethereum'
+import { sendAcknowledgement } from './acknowledgement'
 
-const log = Debug('hopr-core:forward')
+// const log = Debug('hopr-core:forward')
 const FORWARD_TIMEOUT = durations.seconds(6)
-const ACKNOWLEDGEMENT_TIMEOUT = durations.seconds(2)
 
-class PacketForwardInteraction {
+export class PacketForwardInteraction {
   private mixer: Mixer
-  private concurrencyLimiter
 
   constructor(
-    private db: HoprDB,
-    private paymentChannels: HoprCoreEthereum,
-    private id: PeerId,
-    private libp2p: LibP2P,
     private subscribe: any,
     private sendMessage: any,
-    private onMessage: (msg: Uint8Array) => void
+    private privKey: PeerId,
+    private chain: HoprCoreEthereum,
+    private emitMessage: (msg: Uint8Array) => void,
+    private db: HoprDB
   ) {
     this.mixer = new Mixer(this.handleMixedPacket.bind(this))
-    this.concurrencyLimiter = oneAtATime()
     this.subscribe(PROTOCOL_STRING, this.handlePacket.bind(this))
   }
 
   async interact(counterparty: PeerId, packet: Packet): Promise<void> {
-    await this.sendMessage(counterparty, PROTOCOL_STRING, packet, {
+    await this.sendMessage(counterparty, PROTOCOL_STRING, packet.serialize(), {
       timeout: FORWARD_TIMEOUT
     })
   }
 
-  async handlePacket(msg: Uint8Array) {
-    const arr = msg.slice()
-    const packet = new Packet(this.libp2p, this.paymentChannels, this.db, this.id, {
-      bytes: arr.buffer,
-      offset: arr.byteOffset
-    })
+  async handlePacket(msg: Uint8Array, remotePeer: PeerId) {
+    const packet = Packet.deserialize(msg.slice(), this.privKey, remotePeer)
 
     this.mixer.push(packet)
   }
 
   async handleMixedPacket(packet: Packet) {
-    const { id, onMessage, sendMessage } = this
-    const interact = this.interact.bind(this)
-    this.concurrencyLimiter(async function () {
-      // See discussion in #1256 - apparently packet.forwardTransform cannot be
```

Comment:
What was the resolution on calling packet.forwardTransform concurrently?

Reply:
Various things:
- using sync crypto instead of their async counterparts
- seperating (async) ticket creation from packet transformation
- general avoidance of async / callback operations